### ISSUE TITLE
Best options for forming a set leveling and raid team?

### ISSUE THREAD

**Issue by Dreathean** [liyunfan1223/mod-playerbots#170]
**Is your feature request related to a problem? Please describe.**

No issue, just wanting to discuss with fellow players to determine the best currently available options to form a set team for leveling and eventually raiding. My goals are a team/guild of 25 (including myself and a friend) with specifically chosen class, names, race, etc that will level as we level and can pull some as needed for dungeons, and then ultimately use the same team for raiding. Persistence in these playerbots is key.

Currently there seem to be three options: 
1) Create alts and allow playerbots to control them via .playerbots bot add (name). This does allow you to automatically create the name, race, and gender that you want. However, unless I'm missing something there are serious drawbacks: limited to 9 alts and they do not get all of the "randombot" abilities like free food, talent choices, automatic gearing while leveling, etc. Currently does not seem feasible for my plans.

2) Use the playerbots bot addclass (class) option. This allows you to generate a full randombot of the class you want, however main drawbacks are you can't choose name, race, or gender. However I suppose you should be able to change all 3 to your liking in the SQL database, correct? Then you can add them all to your friendlist and use the friends command to make them all come online when you wish. Not 100% sure how that command interacts with real friends however, hopefully that doesn't try to make real characters on your friends list activated via playerbots.

3) Basically the same as 2 except use various bots automatically generated in your set amount of randombots and adjust as needed. To be honest not sure of the pros and cons of doing one or the other.

At this time I'm planning on using option 2 and hoping it won't be difficult to adjust the names, race, and gender as desired in the database. Am I missing anything? Appreciate any and all ideas and insight!

**Comment by Dreathean** [comment]
Slight update: in testing option #2, the randombots only seem to want to follow me even after uninviting them from party. If I teleport somewhere else, they just stand where they last were. Tried resetAI but that didn't seem to do anything. Not sure what to do here? Would like them to resume regular randombot activities after uninviting them from party.

**Comment by liyunfan1223** [comment]
My opinion is option 1 should be the best way :D, or combine both options 1 and 2.

You can invite characters from other accounts to the same guild and set `AllowGuildBots=1` to control more than 9 bots.

After inviting them, using the `maintenance` command can automatically make them prepare almost everything before the raid (without initialized equipment), like learning available spells, skills, talents, and glyphs, initializing food and reagents, enchanting and repairing, etc. The `equip upgrade` command makes them choose better items in their bag to equip (this function is a work in progress, they may be unable to choose the best item so far).

And option 2 can be a supplement when we don't have enough characters.

**Comment by Dreathean** [comment]
Good to know about using extra accounts and guild to control more alt-created playerbots!

I forgot about the new maintenance command - that is huge and definitely solves a lot of those issues. And presumably these alt-created playerbots are safe from any potential deletion when reseting randombots. 

However, it seems that this does not include the automatic gearing upon level-up (not factoring in the level cap equipment persistence)? Or will they be able to do that as well? If not we can work around that by just using randombots during leveling and then generate the alt-bots when we're ready to raid, but then that would also still involve gearing them up for starter gear. Which is fine if that is the case, just seeing if it is possible for these alt-bots to gear themselves up to the equipment persistence level cap. It would just be cool to watch the progress on your guildie alt-bots alongside your own leveling and then seamlessly use them to raid you with haha.

**Comment by liyunfan1223** [comment]
Yeah, an easier starter gearing makes sense. Perhaps it's better to implement it as another command to avoid overusing it.

**Comment by Dreathean** [comment]
Yup, the main thing is that it is simply not feasible to manually keep 24 other bots geared up during the leveling process (raiding is of course different and I intend to handle that manually). Would it be possible to implement a command similar to the maintenance command that allows them to use the same auto-gearing process randombots use while leveling up?
